core/src/main/scala/kafka/tools/GetOffsetShell.scala

```diff
@@ -224,9 +252,81 @@ object GetOffsetShell {
   /**
    * Return the partition infos. Filter them with topicPartitionFilter.
    */
-  private def listPartitionInfos(consumer: KafkaConsumer[_, _], topicPartitionFilter: PartitionInfo => Boolean): Seq[PartitionInfo] = {
-    consumer.listTopics.asScala.values.flatMap { partitions =>
-      partitions.asScala.filter(topicPartitionFilter)
+  private def listPartitionInfos(
+    client: Admin,
+    topicPartitionFilter: TopicPartitionFilter,
+    excludeInternalTopics: Boolean
+  ): Seq[TopicPartition] = {
+    val listTopicsOptions = new ListTopicsOptions().listInternal(!excludeInternalTopics)
+    val topics = client.listTopics(listTopicsOptions).names.get
+    val filteredTopics = topics.asScala.filter(topicPartitionFilter.isTopicAllowed)
+
+    client.describeTopics(filteredTopics.asJava).allTopicNames.get.asScala.flatMap { case (topic, description) =>
+      description
+        .partitions
+        .asScala
+        .map(tp => new TopicPartition(topic, tp.partition))
+        .filter(topicPartitionFilter.isTopicPartitionAllowed)
     }.toBuffer
   }
 }
+
+/**
+ * Used to filter partitions after describing them
```

Comment:
Could we remove `after describing them` and have a more general description? Maybe you can put it like `Used to filter partitions certain criteria` or sth else.

Same comment applied to below. Thanks.

Reply:
Yeah, I reworded these annotations.